### DISCUSSION TITLE
DOS gaming console

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#2140] [General]
Please see this funny video about the
DOS gaming console:
https://www.youtube.com/watch?v=EbnDqni5LEk
It seems to be based on FreeDOS.
Doesn't look very functional, but the
guy managed to run a few games after
all. :) Only pc-speaker sound is available,
but today there is a new SB emulator
somewhere on github, which could be
used here.

Hmm, have vague recollections that I've
already seen this... Hope I am not posting
that URL for the second time?

**Comment by jharrison03** [comment]
Useless without comprehensive sound support, I think.

**Reply by stsp** [reply]
Yes, but these days we have this:
https://github.com/crazii/SBEMU
So sound support should be possible.

**Comment by jschwartzenberg** [comment]
It looks like a repurposed thin-client? I've seen videos before of people playing around with FreeDOS on such hardware. I imagine that it will also depend a lot on the graphics chip whether VGA modes (and prior) are handled nicely.

**Reply by stsp** [reply]
CD-MAN seems to work, which is CGA? :)
So probably VGA is rather well there.

**Reply by jschwartzenberg** [reply]
I have no doubt that the pixels actually show up and such. But having it look nice when things move relatively fast is something else. Smooth scrolling, avoiding things like jiggling score boxes (Commander Keen 4/5/6), etc.

**Reply by stsp** [reply]
> Smooth scrolling, avoiding things like jiggling score boxes (Commander Keen 4/5/6), etc.

You've been heard:
https://liliputing.com/past-meets-present-in-this-200-mini-laptop-with-a-intel-8088-chip-and-640kb-of-memory/

Does this work for you? :)
OTOH with this device, it might be a
bit more challenging to download the
needed game from an internet...

**Reply by jschwartzenberg** [reply]
I think we discussed this in the past :)
But see here with regards to the display quality: https://y.com.sb/watch?v=6bODiZ5bP84

**Reply by stsp** [reply]
Oh, CGA, and GPL violations with bios... :(

**Reply by stuaxo** [reply]
Think the later versions rectified the GPL violations.  Still, EGA would be nice if going for a pre 286.

**Reply by stsp** [reply]
I wonder why even selling 8086+CGA.
Most of the DOS games won't work, so
what's the point...
Google reveals even freedos doesn't
boot there (I guess it is built for 186+).

**Reply by ecm-pushbx** [reply]
> Google reveals even freedos doesn't
boot there (I guess it is built for 186+).

The FreeDOS kernel shouldn't need an 186 when built for 8086 target. Besides, I think the Book 8088 (both versions) actually ships with a NEC V20 which supports all 186 extensions to the instruction set.

**Reply by stsp** [reply]
OK, cool, but does freedos boot? :)

**Reply by ecm-pushbx** [reply]
Yes: http://www.bttr-software.de/forum/forum_entry.php?id=21124

**Reply by jschwartzenberg** [reply]
Reading through the forum thread reveals there's a v2 with VGA. Here is a video: https://yt.artemislena.eu/watch?v=Z45rR8tKPEc
It seems things are not always without slight defects though.

**Reply by stsp** [reply]
Cool, they put a much better LCD
too, it seems. So they are really up
to something. :) He even managed
to get USB to work, after updating
the driver...
I haven't realized what windows
he was (unsuccessfully) trying to
install, 3.0 or 3.1?

**Reply by stsp** [reply]
OK, its 3.0, as it seems, the one installed
was with CGA driver, so he tried with VGA...

**Reply by jschwartzenberg** [reply]
I think they dropped the mode needed for these CPUs in Windows 3.1? Only early beta versions of Windows 3.1 would work I think.

**Reply by stsp** [reply]
Yes, but that was windows-3.0 after all,
not 3.1. You can see it from text under
video by clicking "Show more".
In the video chapters you can see:
`[00:42:52] - Windows 3.0 vs VGA/EGA Drivers Part 2`
So the VGA version of windows-3.0
fails to install.

**Reply by jschwartzenberg** [reply]
In the thread that ecm-pushbx linked it seems somebody managed to debug this and they solved the hardware issue together with the supplier.

**Reply by stsp** [reply]
The thread is too long.
Maybe you can point to the particular
msg of an interest.
Also I think ecm's claim that freedos
boots w/o any problems, refers to the
v2 model of the device.
Its always good if they are working on
their product. But I wonder how the
existing users receive an update.
Is there a bios reflash procedure
(huh, on that hw, really?), or should
they always re-buy the device?

**Reply by ecm-pushbx** [reply]
In the opening post, the problem booting FreeDOS also occurs on the Book8088 v2: http://www.bttr-software.de/forum/forum_entry.php?id=21061

Also, someone said they seemed to get the same error condition on 86Box, which runs on Linux so I may soon be able to debug this hang: (If, indeed, it is the same hang as on the Book8088 v2) http://www.bttr-software.de/forum/forum_entry.php?id=21225

**Reply by stsp** [reply]
Interestingly, freedos was not updated
for years, before fdpp appeared.
It was completely dead.
I even went through the freedos bugzilla
and fixed many of the mentioned problems
in fdpp, because that was easy, just no one
ever cared to do that for freedos.
But now it seems everyone is fixing freedos. :)
Everyone is porting fdpp fixes or making a
new ones.
Not sure if its related to fdpp, maybe just a
coincidence. But a rather surprising one. :)
I'd even said its a good one, if not for the
ultimate hostility of the freedos ML in
general, and to fdpp in particular.

**Reply by jschwartzenberg** [reply]
I misremembered. The relevant message on Windows with VGA is actually in a forum thread that's linked from that thread. Here it is:
https://forum.vcfed.org/index.php?threads/book-8088-discovery-and-modification-thread.1245155/post-1355334

**Comment by stuaxo** [comment]
Maybe something like Launchbox for MSDOS would be the thing to put on it
https://forums.launchbox-app.com/files/file/934-launchbox-for-ms-dos-special-edition/

**Reply by stsp** [reply]
This seems to lack the capability of
downloading the DOS games from
internet, unless I am mistaken.

**Reply by stuaxo** [reply]
Yes, that's true.

**Comment by stsp** [comment]
https://www.youtube.com/watch?v=USHvvSbYmJA
Another DOS gaming PC.
This time, looks like with SB and all
the good stuff in. :)

**Reply by stuaxo** [reply]
The weecee is a great project, I wonder if he will be able to bring the price down in future.

Along similar lines, the remakes of Soundblaster and Ultrasound soundcards are interesting - I think there's a really low cost one with PiZero inside that can do both.

**Reply by ecm-pushbx** [reply]
Interesting! Thanks for the link.

**Reply by stsp** [reply]
But as for me, I don't understand why
is it such a big deal to run on a bare h/w,
w/o an emulator.
Anyone could do the same machine,
but with RPi+dosemu2. :)
I really suspect that the only problem
is that people have not yet discovered
the way of booting into an emulator
very quickly, and the emulators themselves
are not fast enough on RPi.
